testdata/testoutput2

```diff
@@ -20646,10 +20765,151 @@ Failed: error 150 at offset 5: invalid range in character class
 No match
 
 /[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[z]&&a]]]]]]]]]]]]]]]/alt_extended_class
-Failed: error 207 at offset 115: character classes are too deeply nested
+Failed: error 207 at offset 115: character class nesting is too deep
 
 /[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a[b]&&a[a&&a[z]]]]]]]]]]]]]]]]/alt_extended_class
-Failed: error 207 at offset 118: character classes are too deeply nested
+Failed: error 207 at offset 118: character class nesting is too deep
+
+# --------------
+
+# EXTENDED CHARACTER CLASSES (Perl)
+
+# allow-empty-class does nothing inside (?[...])
+/(?[ []] ])/allow_empty_class
+    ]
+ 0: ]
+
+# bad-escape-is-literal does nothing inside (?[...])
+/[ \j ]/
+Failed: error 103 at offset 3: unrecognized character follows \
+
+/(?[ \j ])/
+Failed: error 103 at offset 5: unrecognized character follows \
+
+/[ \j ]/bad_escape_is_literal
+    j
+ 0: j
+\= Expect no match
+    k
+No match
+
+/(?[ \j ])/bad_escape_is_literal
+Failed: error 103 at offset 5: unrecognized character follows \
+
+/(?[ [\j] ])/bad_escape_is_literal
+Failed: error 103 at offset 6: unrecognized character follows \
+
+/(?[ (\j) ])/bad_escape_is_literal
+Failed: error 103 at offset 6: unrecognized character follows \
+
+# We can't test error cases in testinput1
+
+/(?[])/
+Failed: error 214 at offset 4: empty expression in character class
+
+/(?[/
+Failed: error 106 at offset 3: missing terminating ] for character class
+
+/(?[]/
+Failed: error 214 at offset 4: empty expression in character class
+
+/(?[\n/
+Failed: error 106 at offset 5: missing terminating ] for character class
+
+/(?[\n]/
+Failed: error 215 at offset 6: unexpected ] with no following parenthesis in (?[...]
```

Comment:
`(?[` or `(?[...])`

Reply:
It's intentional. I'm exercising not just every line of code, but also every branch in all the `ifs`. So there are lots of tests where I take valid patterns, and chop bits off the end, in order to exercise the `if (ptr < ptrend && ...)` part of the conditions.

Reply:
I mean the error message should be changed. `(?[...])` used most of the time

Reply:
I see. I was following Perl's error message, which is:

```sh
$ echo '/(?[\n]/;' | perl
Unexpected ']' with no following ')' in (?[... in regex; marked by <-- HERE in m/(?[\n] <-- HERE / at - line 1.
```

I just tweaked the message a tiny bit. The `in (?[...]` (or "in foo" in all error messages) was supposed to mean "in this text which you provided", and indeed `(?[...]` is what the user provided.

I'll change it to `(?[...` to exactly match Perl.